src/InverseSolve/computeGradMisfit.jl

```diff
@@ -80,67 +83,67 @@ function computeGradMisfit(sigma,
 	end
 	workerList = unique(workerList)
 	# send sigma to all workers
-	sigmaRef = Array(RemoteRef,maximum(workers()))
-	dFiRef   = Array(RemoteRef,maximum(workers()))
+	sigmaRef = Array(Future,maximum(workers()))
+	dFiRef   = Array(RemoteChannel,maximum(workers()))
 	dF = zeros(length(sigma))
 
 	commTime = 0.0
 	compTime = 0.0
 	updateTimes(c1,c2) = (commTime+=c1; compTime+=c2)
 	updateDF(x) = (dF+=x)
-	
-	
+
+
 	tic()
 	@sync begin
 		for p = workers()
 				@async begin
 					# send model to worker and get a remote ref
 					tic()
-						sigmaRef[p] = remotecall(p,identity,sigma)   # send sigma to workers
-						dFiRef[p] = remotecall(p,identity,0.0) # get remote Ref to part of gradient
+						sigmaRef[p] = remotecall(identity,p,sigma)   # send sigma to workers
+						  dFiRef[p] = initRemoteChannel(identity,p,zeros(length(sigma))) # get remote Ref to part of gradient
 					c1 = toq()
 					updateTimes(c1,0.0)
-					
+
 					# the actual computation
 					for idx=indFors
 						if pMisRefs[idx].where==p
-							isDone,c1,c2 = remotecall_fetch(p, computeGradMisfit,sigmaRef[p],DcRef[idx],pMisRefs[idx],dFiRef[p])
+							isDone,c1,c2 = remotecall_fetch(computeGradMisfit,p, sigmaRef[p],DcRef[idx],pMisRefs[idx],dFiRef[p])
 							updateTimes(c1,c2)
 						end
-					end	
-					
+					end
+
 					# fetch result and add
 					tic()
-					updateDF(fetch(dFiRef[p]))				
+					updateDF(fetch(dFiRef[p]))
 					c1 = toq()
 					updateTimes(c1,0.0)
 				end
-		end 
+		end
 	end
 	chkTime=toq()
-	
+
 	totalTime = toq()
 	return dF
 end
 
-# function computeGradMisfit(mc,Dc::Array,pMis::Array{RemoteRef},M2M::Array{RemoteRef},indFors=1:length(PF))
+# function computeGradMisfit(mc,Dc::Array,pMis::Array{Future},M2M::Array{Future},indFors=1:length(PF))
```

Comment:
Henh. That's interesting, where did it get called from? Not a problem to leave the commented out code there as is. The method seems to be broken as written however. It looks like someone started to update it to use the misfitParam, stopped before they finished and then commented out the code. I suppose the code should either be deleted or fixed and uncommented. Is it meant to be a fault tolerant version of the method just above it?